esrally/mechanic/telemetry.py

```diff
@@ -276,6 +276,159 @@ def detach_from_node(self, node, running):
+                        "specified in --target-hosts but it had [{}].".format(cluster_name))
+            self.specified_cluster_names = self.indices_per_cluster.keys()
+
+        self.metrics_store = metrics_store
+        self.samplers = []
+
+    def attach_to_cluster(self, cluster):
+        # This cluster parameter does not correspond to the cluster names passed in target.hosts, see on_benchmark_start()
+        super().attach_to_cluster(cluster)
+
+    def on_benchmark_start(self):
+        recorder = []
+        for cluster_name in self.specified_cluster_names:
+            recorder = CCRStatsRecorder(cluster_name, self.clients[cluster_name], self.metrics_store, self.sample_interval,
+                                        self.indices_per_cluster[cluster_name] if self.indices_per_cluster else None)
+            sampler = SamplerThread(recorder)
+            self.samplers.append(sampler)
+            sampler.setDaemon(True)
+            # we don't require starting recorders precisely at the same time
+            sampler.start()
+
+    def on_benchmark_stop(self):
+        if self.samplers:
+            for sampler in self.samplers:
+                sampler.finish()
+
+
+class CCRStatsRecorder:
+    """
+    Collects and pushes CCR stats for the specified cluster to the metric store.
+    """
+
+    def __init__(self, cluster_name, client, metrics_store, sample_interval, indices=None):
+        """
+        :param cluster_name: The cluster_name that the client connects to, as specified in target.hosts.
+        :param client: The Elasticsearch client connection to the cluster.
+        :param metrics_store: The configured metrics store we write to.
+        :param sample_interval: integer controlling the interval, in seconds, between collecting samples.
+        :param indices: optional list of indices to filter results from.
+        """
+
+        self.cluster_name = cluster_name
+        self.client = client
+        self.metrics_store = metrics_store
+        self.sample_interval= sample_interval
+        self.indices = indices
+
+    def __str__(self):
+        return "ccr stats"
+
+    def record(self):
+        """
+        Collect CCR stats for indexes (optionally) specified in telemetry parameters and push to metrics store.
+        """
+
+        # ES returns all stats values in bytes or ms via "human: false"
+        stats = self.client.transport.perform_request('GET', '/_xpack/ccr/_stats', params={"human": "false"})
+        if self.indices:
+            # Record metrics only for indices specified with telemetry-params.
+            for index in self.indices:
+                try:
+                    if stats[index] == {}:
+                        continue
+                    self.record_all_index_stats(index, stats[index])
+                except KeyError:
+                    logger.warning("The telemetry parameter 'ccr-stats-indices' specified the index [%s] for cluster [%s] but there are "
```

Comment:
That's right, I am finding this useful because for CCR we aren't setting up the cluster with Rally and rely on custom runners to make an index start following an index in another already joined cluster.

Reply:
Ok.